### ISSUE TITLE
Conversion error?

### ISSUE THREAD

**Issue by andreasvarga** [JuliaMath/Polynomials.jl#316]
For the next conversion attempt it is OK to get an error:

```
Polynomial{Int,:z}([1+im,2])
ERROR: InexactError: Int64(1 + 1im)
Stacktrace:
 [1] Real at .\complex.jl:37 [inlined]
 [2] convert at .\number.jl:7 [inlined]
 [3] setindex! at .\array.jl:847 [inlined]
 [4] Polynomial{Int64,:z}(::Array{Complex{Int64},1}) at C:\Users\Andreas\.julia\packages\Polynomials\1nhQ0\src\polynomials\Polynomial.jl:43        
 [5] top-level scope at REPL[20]:1
```
and the same error is to be expected for the next one
```
julia> Polynomial{Int,:z}(1.1+im)
Polynomial(1.1 + im)
```
But there is no error here (and in many similar cases)!

**Comment by jverzani** [comment]
Yeah, that is a bug. Thanks. When the constructor specifies a type, it should throw an error if it can't be done.

**Comment by jverzani** [comment]
Closed by #317
